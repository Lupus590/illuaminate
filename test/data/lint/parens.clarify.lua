-- config: (lint (only (syntax:redundant-parens)) (allow-clarifying-parens true))

local _ = (2 + 2)
local _ = (2 + 2)() -- ok
local _ = (2 + 2)[1] -- ok

local _ = (2 + 2) * 2 -- ok
local _ = 2 + (2 * 2) -- ok
local _ = 2 + (2 + 2) -- ok
local _ = (2 + 2) + 2 -- ok

local _ = ((2 + 2)) * 2
local _ = 2 + ((2 * 2))
local _ = 2 + ((2 + 2))
local _ = ((2 + 2)) + 2
