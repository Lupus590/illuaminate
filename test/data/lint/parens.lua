-- config: (lint (only syntax:redundant-parens))

-- Redundant parenthesis around references
do
  (f)((f), {
      (f)
  })
end

-- Redundant parenthesis around literals
do
  (nil)() ; -- ok
  ((nil))()
  local _ = (nil)[1] -- ok
  local _ = ((nil))[1]
end


-- Redundant parenthesis around calls
do
  local _, _ = (f()) -- ok
  local _ = (f())

  f((f())) -- ok
  f((f()), 0)

  local _ = { (f()) } -- ok
  local _ = { (f()), 0 }

  local function _() return (f()) end -- ok
  local function _() return (f()), 0 end
end

-- Redundant parenthesis around dots
do
  local _, _ = (...) -- ok
  local _ = (...)

  f((...)) -- ok
  f((...), 0)

  local _ = { (...) } -- ok
  local _ = { (...), 0 }

  local function _(...) return (...) end -- ok
  local function _(...) return (...), 0 end
  -- TODO: local function _(...) return(x)end
end

-- Operators
do
  local _ = (2 + 2)
  local _ = (2 + 2)() -- ok
  local _ = (2 + 2)[1] -- ok

  local _ = (#2)
  local _ = (#2)() -- ok
  local _ = (#2)[1] -- ok

  local _ = (2 + 2) * 2 -- ok
  local _ = 2 + (2 * 2)
  local _ = 2 + (2 + 2) -- ok
  local _ = 2 * (2 + 2) + 1 -- ok
  local _ = (2 + 2) + 2

  local _ = not(x)
  local _ = (x)and(x)
end
