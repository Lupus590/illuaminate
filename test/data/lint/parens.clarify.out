parens.clarify.lua:[3:10-3:17]: Unnecessary parenthesis. [syntax:redundant-parens]
   │
3  │ local _ = (2 + 2)
   │           ^^^^^^^
parens.clarify.lua:[12:10-12:19]: Unnecessary parenthesis. [syntax:redundant-parens]
    │
12  │ local _ = ((2 + 2)) * 2
    │           ^^^^^^^^^
parens.clarify.lua:[13:14-13:23]: Unnecessary parenthesis. [syntax:redundant-parens]
    │
13  │ local _ = 2 + ((2 * 2))
    │               ^^^^^^^^^
parens.clarify.lua:[14:14-14:23]: Unnecessary parenthesis. [syntax:redundant-parens]
    │
14  │ local _ = 2 + ((2 + 2))
    │               ^^^^^^^^^
parens.clarify.lua:[15:10-15:19]: Unnecessary parenthesis. [syntax:redundant-parens]
    │
15  │ local _ = ((2 + 2)) + 2
    │           ^^^^^^^^^
Compilation succeeded with 5 warnings
@@ -1,6 +1,6 @@
 -- config: (lint (only (syntax:redundant-parens)) (allow-clarifying-parens true))
 
-local _ = (2 + 2)
+local _ = 2 + 2
 local _ = (2 + 2)() -- ok
 local _ = (2 + 2)[1] -- ok
 
@@ -9,8 +9,8 @@
 local _ = 2 + (2 + 2) -- ok
 local _ = (2 + 2) + 2 -- ok
 
-local _ = ((2 + 2)) * 2
-local _ = 2 + ((2 * 2))
-local _ = 2 + ((2 + 2))
-local _ = ((2 + 2)) + 2
+local _ = (2 + 2) * 2
+local _ = 2 + (2 * 2)
+local _ = 2 + (2 + 2)
+local _ = (2 + 2) + 2
 
